mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -129,7 +133,7 @@
           // TODO: We do not support residual evaluation for HIVE and PIG in memory data model yet
           checkResiduals(task);
         }
-        splits.add(new IcebergSplit(conf, task));
+        splits.add(new IcebergSplit(conf, task, table.io(), table.encryption()));
```

Comment:
While I would like to get the encryption manager and io changes in, I don't think that they should be mixed into this commit. Was it necessary to do this for some reason?

Reply:
The `GenericDeleteFilter` constructor needs `FileIO` as parameter.

Reply:
In that case, this can pass the `FileIO` somehow, or we can work on getting the other PR done before this one. But I don't think we should mix the two features together.

Reply:
For now, this could create a new `HadoopFileIO` and use that instead. That would be the easiest path forward.

Reply:
@holdenk just added a `FileIO` instance to this class, so you can use that instead of mixing the two PRs together. Thanks @holdenk!

Reply:
Thank you @holdenk @rdblue. I updated this.